python/graphstorm/model/utils.py

```diff
@@ -681,6 +688,57 @@ def save_prediction_results(predictions, prediction_path, rank):
 
     th.save(predictions, os.path.join(prediction_path, f"predict-{pad_file_index(rank)}.pt"))
 
+def save_node_prediction_results(predictions, prediction_path):
+    """ Save node predictions to the given path
+
+        Parameters
+        ----------
+        prediction: tensor
+            The dict of tensors of predictions.
+        prediction_path: str
+            The path of the prediction is saved.
+    """
+    rank = get_rank()
+    world_size = get_world_size()
+    for ntype, pred in predictions.items():
+        save_prediction_results(pred,
+                                os.path.join(prediction_path, ntype),
+                                rank)
+    if rank == 0:
+        meta_fname = os.path.join(prediction_path, "result_info.json")
+        meta_info = {
+            "format": "pytorch",
+            "world_size": world_size,
+            "ntypes": list(predictions.keys())
+        }
```

Comment:
why is this useful?

Reply:
format and world_size are useful for remapping the id into raw ID space.
ntypes is recording which node type(s) have prediction results. the remapping script can use it to do some sanity checks.